fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -115,20 +120,68 @@ const socialSlice = createSlice({
         chirp: ChirpState,
       }>) {
         const { laoId, chirp } = action.payload;
+
+        if (!(laoId in state.byLaoId)) {
+          state.byLaoId[laoId] = {
+            allIdsInOrder: [],
+            byId: {},
+            byUser: {},
+            reactionsByChirp: {},
+          };
+        }
+
         const store = state.byLaoId[laoId];
 
-        if (store.byId[chirp.id]) {
-          if (!store.byId[chirp.id].isDeleted) {
-            store.byId[chirp.id] = chirp;
-          }
+        // store the deleted chirp
+        store.byId[chirp.id] = new Chirp({
```

Comment:
and I believe here you're not doing the corresponding check on the sender, to make sure that only the original sender can delete ?